src/game.cpp

```diff
@@ -11762,16 +11762,32 @@ void game::update_overmap_seen()
 {
     const tripoint ompos = u.global_omt_location();
     const int dist = u.overmap_sight_range( light_level( u.posz() ) );
+    const int dist_squared = dist * dist;
     // We can always see where we're standing
     overmap_buffer.set_seen( ompos.x, ompos.y, ompos.z, true );
-    for( int x = ompos.x - dist; x <= ompos.x + dist; x++ ) {
-        for( int y = ompos.y - dist; y <= ompos.y + dist; y++ ) {
+    for( int dx = -dist; dx <= dist; dx++ ) {
+        for( int dy = -dist; dy <= dist; dy++ ) {
+            if( trigdist && dx * dx + dy * dy > dist_squared ) {
+                continue;
+            }
+            int x = ompos.x + dx;
+            int y = ompos.y + dy;
+            float multiplier;
+            if( trigdist ) {
+                float angle = fmod( atan2( dy, dx ) + M_PI, M_PI / 2 );
+                if( angle > M_PI / 4 ) {
+                    angle = M_PI / 2 - angle;
+                }
+                multiplier = 1 / cos( angle );
```

Comment:
You probably don't need any of these trigonometric functions and angle adjustments. The math here is fairly simple:
```c++

const int h_squared = dx * dx + dy * dy;

if( h_squared == 0 || h_squared > dist_squared )
    continue;
    
const float multiplier = std::sqrt( h_squared ) / std::max<float>( std::abs( dx ), std::abs( dy ) );
```

Reply:
I guess it is just that simple, huh... I'm going to do another overmap vision PR soon; I'm happy to sneak this refactor in with that.